workload/src/main/java/com/google/finapp/WorkloadClient.java

```diff
@@ -14,28 +14,91 @@
 
 package com.google.finapp;
 
+import com.google.common.collect.ImmutableList;
+import com.google.finapp.CreateAccountRequest.Status;
 import com.google.finapp.FinAppGrpc.FinAppBlockingStub;
 import com.google.protobuf.ByteString;
 import io.grpc.ManagedChannel;
 import io.grpc.StatusRuntimeException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-/** A gRPC client for the finance sample app. */
-public class WorkloadClient {
+/** A gRPC client for the finance sample app. Performs a series of tasks (gRPC methods) in order. */
+public class WorkloadClient implements Runnable {
 
   private final FinAppBlockingStub blockingStub;
   private static final Logger logger = Logger.getLogger(WorkloadClient.class.getName());
+  private final ImmutableList<Task> tasks;
+  private final List<ByteString> ids;
+  private final Random random = new Random();
 
-  private WorkloadClient(ManagedChannel channel) {
+  private WorkloadClient(ManagedChannel channel, ImmutableList<Task> tasks) {
```

Comment:
I wanted to pass immutable objects to this client since it needs to be threadsafe. I now realize that the client is instantiated before passed to a thread, so this conversion from list to immutable list can happen during the client instantiation.